dts/arm/cypress/psoc6.dtsi

```diff
@@ -63,6 +64,165 @@
 	};
 
 	soc {
+		hsiom: hsiom@40310000 {
+			compatible = "cypress,psoc6-hsiom";
+			reg = <0x40310000 0x2024>;
+			interrupts = <15 1>, <16 1>;
+			label = "HSIOM";
+			status = "disabled";
+		};
+
+		p0: gpio@40320000 {
```

Comment:
The datasheet and header files seem to refer to a module instance as `GPIO_PRTx`. Maybe it would be better to use `gpio_prt0` for node name instead of `p0`? While short `p0` is a bit cryptic. When looking e.g. at the board dts files it's not obvious which hardware module the label refers to.

Reply:
Cypress defined two ways to denominate a **port**:

![image](https://user-images.githubusercontent.com/7027887/100624301-10382c00-3302-11eb-866c-91aff627695f.png)

At first,  **device pin** denomination seems be the right move, Looking at documentation from a serial on port 9:

![image](https://user-images.githubusercontent.com/7027887/100625351-2b576b80-3303-11eb-903e-e478a164a175.png)

At development kits, they are using the shortened form too:

![image](https://user-images.githubusercontent.com/7027887/100625004-cbf95b80-3302-11eb-9850-cc3d46bdcdfb.png)

This lead us to three options:

gpios/pinctrl-0 using shortened device pin denomination (actual):
```
		user_led: led_0 {
			gpios = <&p13 7 GPIO_ACTIVE_HIGH>;
		};
&uart2 {
       pinctrl-0 = <&p9_0_uart2_rx &p9_1_uart2_tx>;
 };
```

gpios/pinctrl-0 using shortened port register denomination (new):
```
		user_led: led_0 {
			gpios = <&prt13 7 GPIO_ACTIVE_HIGH>;
	};
&uart2 {
       pinctrl-0 = <&prt9_0_uart2_rx &prt9_1_uart2_tx>;
 };
```

gpios/pinctrl-0 using full port register denomination (new):
```
		user_led: led_0 {
			gpios = <&gpio_prt13 7 GPIO_ACTIVE_HIGH>;
		};
&uart2 {
       pinctrl-0 = <&gpio_prt9_0_uart2_rx &gpio_prt9_1_uart2_tx>;
 };
```

I agreed that it will be better follow PSoC-6 Architecture and Register TRMs (Technical Reference Manual). That documentation are following port register denomination. 

As was wondering if you are willing to accept the **prt** instead **gpio_prt**. The full port register denomination seems redundant when declaring **gpios** and odd at **pinctrl**. It is important mention that GPIO block will be inside a pinctrl block soon #28645 (out of date).

Reply:
Thanks @nandojve for the explanation and sorry for the late answer. My comment related strictly to the name of GPIO module. To refer to a pin we could use another / shorthand notation. So far node labels in DTS followed naming used by the vendor in the datasheet. `gpio_prt0` would follow the existing convention, even if it is the longest. In any case, I'm not going to block this PR if you prefer to stay with a shorter notation.

Reply:
Hi @mnkp ,

I was thinking more about it and agreed with you. The current naming convention will be better.